src/Plugin/authentication/OAuth2ServerAuthentication.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Drupal\restful\Plugin\authentication;
+
+use Drupal\Component\Plugin\PluginBase;
+use Drupal\restful\Exception\UnauthorizedException;
+use Drupal\restful\Http\RequestInterface;
+use Drupal\restful\Plugin\ResourcePluginManager;
+
+/**
+ * Authentication support for oauth2_server.
+ *
+ * @Authentication(
+ *   id = "oauth2_auth",
```

Comment:
I was using `basic_auth` as an example. I see other plugins don't follow this convention. Changing...